src/Compilers/CSharp/Test/Emit/CodeGen/FixedSizeBufferTests.cs

```diff
@@ -40,38 +40,311 @@ static void Main()
+  IL_001c:  stind.i4
+  IL_001d:  ldloc.1
+  IL_001e:  ldc.i4.4
+  IL_001f:  add
+  IL_0020:  ldloc.1
+  IL_0021:  ldind.i4
+  IL_0022:  stind.i4
+  IL_0023:  ldloca.s   V_0
+  IL_0025:  ldflda     ""int* S.x""
+  IL_002a:  ldflda     ""int S.<x>e__FixedBuffer.FixedElementField""
+  IL_002f:  ldc.i4.4
+  IL_0030:  add
+  IL_0031:  ldind.i4
+  IL_0032:  call       ""void System.Console.WriteLine(int)""
+  IL_0037:  ret
+}");
+        }
+
+        [Fact]
+        public void SimpleFixedBufferNestedField()
+        {
+            var text =
+@"
+using System;
+unsafe struct S
+{
+    public fixed int x[10];
+}
+
+struct  S1
+{
+    public S field;
+}
+
+class Program
+{
+    unsafe static void Main()
+    {
+        S1 c = new S1();
+        c.field.x[0] = 12;
+        ref int i = ref c.field.x[0];
+        Console.WriteLine(i);
+    }
+}
+";
+            CompileAndVerify(text, options: TestOptions.UnsafeReleaseExe, expectedOutput: "12", verify: Verification.Passes)
+                .VerifyIL("Program.Main", @"
+{
+  // Code size       52 (0x34)
+  .maxstack  2
+  .locals init (S1 V_0) //c
+  IL_0000:  ldloca.s   V_0
+  IL_0002:  initobj    ""S1""
+  IL_0008:  ldloca.s   V_0
+  IL_000a:  ldflda     ""S S1.field""
+  IL_000f:  ldflda     ""int* S.x""
+  IL_0014:  ldflda     ""int S.<x>e__FixedBuffer.FixedElementField""
+  IL_0019:  ldc.i4.s   12
+  IL_001b:  stind.i4
+  IL_001c:  ldloca.s   V_0
+  IL_001e:  ldflda     ""S S1.field""
+  IL_0023:  ldflda     ""int* S.x""
+  IL_0028:  ldflda     ""int S.<x>e__FixedBuffer.FixedElementField""
+  IL_002d:  ldind.i4
+  IL_002e:  call       ""void System.Console.WriteLine(int)""
+  IL_0033:  ret
+}");
+        }
+
+        [Fact]
+        public void SimpleFixedBufferNestedFieldClass()
+        {
+            var text =
+@"
+using System;
+unsafe struct S
+{
+    public fixed int x[10];
+}
+
+class  S1
+{
+    public S field;
+}
+
+class Program
+{
+    unsafe static void Main()
+    {
+        S1 c = new S1();
+        c.field.x[0] = 12;
+        ref int i = ref c.field.x[0];
+        Console.WriteLine(i);
+    }
+}
+";
+            CompileAndVerify(text, options: TestOptions.UnsafeReleaseExe, expectedOutput: "12", verify: Verification.Passes)
+                .VerifyIL("Program.Main", @"
+{
+  // Code size       46 (0x2e)
+  .maxstack  3
+  IL_0000:  newobj     ""S1..ctor()""
+  IL_0005:  dup
+  IL_0006:  ldflda     ""S S1.field""
+  IL_000b:  ldflda     ""int* S.x""
+  IL_0010:  ldflda     ""int S.<x>e__FixedBuffer.FixedElementField""
+  IL_0015:  ldc.i4.s   12
+  IL_0017:  stind.i4
+  IL_0018:  ldflda     ""S S1.field""
+  IL_001d:  ldflda     ""int* S.x""
+  IL_0022:  ldflda     ""int S.<x>e__FixedBuffer.FixedElementField""
+  IL_0027:  ldind.i4
+  IL_0028:  call       ""void System.Console.WriteLine(int)""
+  IL_002d:  ret
+}");
+        }
+
+        //PROTOTYPE: I have verified that this passes with the fix in https://github.com/dotnet/roslyn/pull/24909 that has not yet merged
```

Comment:
I just want to make sure that the test is enabled before merging.
will add a feature name

---
In reply to: [170482007](https://github.com/dotnet/roslyn/pull/24966#discussion_r170482007) [](ancestors = 170482007)

Reply:
The linked github PR was since merged.

---
In reply to: [171149585](https://github.com/dotnet/roslyn/pull/24966#discussion_r171149585) [](ancestors = 171149585,170482007)